packages/headless/src/features/configuration/configuration-actions.ts

```diff
@@ -57,6 +70,35 @@ export const updateSearchConfiguration = createAction(
     })
 );
 
+export interface UpdateAnalyticsConfigurationActionCreatorPayload {
+  /**
+   * Whether to enable usage analytics tracking.
+   */
+  enabled?: boolean;
+
+  /**
+   * The origin level 2 usage analytics event metadata whose value should typically be the identifier of the tab from which the usage analytics event originates (e.g., `All`).
+   */
+  originLevel2?: string;
+
+  /**
+   * The origin level 3 usage analytics event metadata whose value should typically be the URL of the page that linked to the search interface that’s making the request (e.g., `https://connect.coveo.com/s/`).
+   */
+  originLevel3?: string;
+
+  /**
+   * The Usage Analytics API base URL to use (e.g., `https://platform.cloud.coveo.com/rest/ua`).
+   */
+  apiBaseUrl?: string;
+
+  /**
+   * The Coveo analytics runtime to use, see https://github.com/coveo/coveo.analytics.js for more info.
+   */
+  runtimeEnvironment?: AnalyticsRuntimeEnvironment;
+}
+
+export type AnalyticsRuntimeEnvironment = IRuntimeEnvironment;
```

Comment:
I'm storing the interface in a type alias here to avoid needing to export `IRuntimeEnvironment` and it's nested interfaces (`WebStorage`, `AnalyticsRequestClient`), which come form the coveo.analytics package.